R/group_data.R

```diff
@@ -380,6 +380,8 @@ trtAssign <- function(dtName, nTrt = 2, balanced = TRUE,
     dt[, `:=`(.stratum = NULL, .n = NULL)]
 
     if (nTrt == 2) dt[, grpExp := grpExp - 1]
+    dt[, grpExp := as.integer(grpExp)]
```

Comment:
Looking at the code when `balanced = FALSE` the `grpExp` are generated via C++ and that returns integers anyway?

Reply:
Yes - `trtObserve` is fine for the reason you stated.